pkg/storage/internal/actions/abort.go

```diff
@@ -0,0 +1,83 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/logging"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/storage/entity"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+)
+
+type abortAction struct {
+	logger           *slog.Logger
+	transactionsRepo TransactionsRepo
+}
+
+func newAbortAction(logger *slog.Logger, transactions TransactionsRepo) *abortAction {
+	return &abortAction{
+		logger:           logging.Child(logger, "abortAction"),
+		transactionsRepo: transactions,
+	}
+}
+
+func (a *abortAction) AbortAction(ctx context.Context, userID int, args *wdk.AbortActionArgs) (*wdk.AbortActionResult, error) {
+	txEntity, err := a.transactionsRepo.FindTransactionByReference(ctx, userID, args.Reference)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find unique transaction by reference: %w", err)
+	}
+
+	if txEntity == nil && a.isPotentiallyTxID(args.Reference) {
+		txEntity, err = a.transactionsRepo.FindTransactionByUserIDAndTxID(ctx, userID, args.Reference)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find transaction by txid: %w", err)
+		}
+	}
+
+	if txEntity == nil {
+		return nil, fmt.Errorf("failed to find unique transaction by reference: expected exactly one transaction with reference %s, found 0", args.Reference)
+	}
+
+	err = a.validateTx(txEntity)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.transactionsRepo.AbortTransactionAtomic(
+		ctx,
+		txEntity.ID,
+		txEntity.TxID,
+		args.Reference,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to abort transaction: %w", err)
+	}
+
+	return &wdk.AbortActionResult{
+		Aborted: true,
+	}, nil
+}
```

Comment:
```suggestion
func (a *abortAction) AbortAction(ctx context.Context, userID int, args *wdk.AbortActionArgs) (*wdk.AbortActionResult, error) {
	txEntity, err := a.transactionsRepo.FindTransactionByReference(ctx, userID, args.Reference)
	if err != nil {
		return nil, fmt.Errorf("failed to find unique transaction by reference: %w", err)
	}

	if txEntity == nil && a.isPotentiallyTxID(args.Reference) {
		txEntity, err = a.transactionsRepo.FindTransactionByUserIDAndTxID(ctx, userID, args.Reference)
		if err != nil {
			return nil, fmt.Errorf("failed to find transaction by txid: %w", err)
		}
	}

	if txEntity == nil {
		return nil, fmt.Errorf("no transaction found with reference or txid %q", args.Reference)
	}

	if err := a.validateTx(txEntity); err != nil {
		return nil, fmt.Errorf("transaction validation failed: %w", err)
	}

	if err := a.transactionsRepo.AbortTransactionAtomic(ctx, txEntity.ID, txEntity.TxID, args.Reference); err != nil {
		return nil, fmt.Errorf("failed to abort transaction: %w", err)
	}

	return &wdk.AbortActionResult{Aborted: true}, nil
}
``` 
or maybe like this WDYT ?

Reply:
i applied your suggestion